src/CentrifugeRouter.sol

```diff
@@ -9,6 +9,11 @@ import {IERC20} from "src/interfaces/IERC20.sol";
 import {ICentrifugeRouter} from "src/interfaces/ICentrifugeRouter.sol";
 import {IPoolManager} from "src/interfaces/IPoolManager.sol";
 
+interface GatewayLike {
+    function topUp() external payable;
```

Comment:
I would call topup prepay & use the balance in case the tx is coming from a prepaid source